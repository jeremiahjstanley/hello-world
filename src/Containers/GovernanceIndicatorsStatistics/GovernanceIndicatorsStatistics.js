import React from 'react';
import { connect } from 'react-redux';
import { LineChart, Line, XAxis, YAxis, Tooltip, Legend, BarChart, Bar } from 'recharts';


export const GovernanceIndicatorsStatistics = (props) => {
  return (
    <div className="statistics-container">
      <LineChart width={900} height={300} data={props.locationData} margin={{ top: 5, right: 20, bottom: 5, left: 0 }}>
        <Line type="monotone" dataKey="Percentile Rank" stroke="#8884d8" />
        <XAxis dataKey="name" />
        <YAxis />
        <Tooltip/>
        <Legend />
      </LineChart>
      <div className="subordinate-graphs">
        <BarChart width={300} height={150} data={props.locationData}>
          <Bar dataKey="Number of Sources" fill='#8884d8'/>
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip/>
          <Legend />
        </BarChart>
        <LineChart width={300} height={150} data={props.locationData} margin={{ top: 5, right: 20, bottom: 5, left: 0 }}>
          <Line type="monotone" dataKey="Standard Error" stroke="#8884d8" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip/>
          <Legend />
        </LineChart>
        <LineChart width={300} height={150} data={props.locationData} margin={{ top: 5, right: 20, bottom: 5, left: 0 }}>
          <Line type="monotone" dataKey="Estimate" stroke="#8884d8" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip/>
          <Legend />
        </LineChart>
      </div>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  locationData: state.locationData
});

export default connect(mapStateToProps)(GovernanceIndicatorsStatistics);