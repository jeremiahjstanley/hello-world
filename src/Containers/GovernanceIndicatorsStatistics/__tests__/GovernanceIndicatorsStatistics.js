import React from 'react';
import { shallow } from 'enzyme';
import { GovernanceIndicatorsStatistics, mapStateToProps } from '../GovernanceIndicatorsStatistics';


describe('GovernanceIndicatorsStatistics', () => {

  let wrapper;
  let mockDataBase;
  let mockDataSet;
  let mockLocation;
  let mockLocationData;

  beforeEach(() => {

    mockDataBase = {
  		name: 'Worldwide Governance Indicators',
  		database_code: 'WWGI'
  	};

    mockDataSet = {
      name: 'Voice and Accountability', 
      dataset_code: 'VA'
    };

    mockLocation = [{
      name:'Burkina Faso',
      alpha_2:'BF',
      alpha_3:'BFA',
      country_code:'854',
      iso_3166_2:'ISO 3166-2: BF',
      region:'Africa',
      sub_region:'Sub-Saharan Africa',
      intermediate_region:'Western Africa',
      region_code:'002',sub_region_code:'202',
      intermediate_region_code:'011'
    }];

    mockLocationData = [];
    // error is here.

    wrapper = shallow(
    	<GovernanceIndicatorsStatistics
        dataBase={mockDataBase}
        dataSet={mockDataSet}
        location={mockLocation}
        locationData={mockLocationData}
    	/>
  	 );

  });

	it('should match the snapshot when the locationData array has length ', () => {

		expect(wrapper).toMatchSnapshot();

	});

  it('should match the snapshot when the locationData array does not have length ', () => {

    wrapper = shallow(
      <GovernanceIndicatorsStatistics 
        dataBase={mockDataBase}
        dataSet={mockDataSet}
        location={mockLocation}
        locationData={[]}
      />
     );

    expect(wrapper).toMatchSnapshot();

  });

	describe('mapStateToProps', () => {

    it('should return a props object with an dataBase, dataSet, location, and locationData property', () => {

      const mockState = {
        dataBase: mockDataBase,
        dataSet: mockDataSet,
        location: mockLocation,
        locationData: mockLocationData
      };

      const expected = mockState;

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);

    });
		
	});
	
});