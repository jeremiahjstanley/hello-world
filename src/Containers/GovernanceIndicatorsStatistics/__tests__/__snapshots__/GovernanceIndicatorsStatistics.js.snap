// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovernanceIndicatorsStatistics should match the snapshot when the locationData array does not have length  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GovernanceIndicatorsStatistics
    dataBase={
        Object {
            "database_code": "WWGI",
            "name": "Worldwide Governance Indicators",
          }
    }
    dataSet={
        Object {
            "dataset_code": "VA",
            "name": "Voice and Accountability",
          }
    }
    location={
        Array [
            Object {
              "alpha_2": "BF",
              "alpha_3": "BFA",
              "country_code": "854",
              "intermediate_region": "Western Africa",
              "intermediate_region_code": "011",
              "iso_3166_2": "ISO 3166-2: BF",
              "name": "Burkina Faso",
              "region": "Africa",
              "region_code": "002",
              "sub_region": "Sub-Saharan Africa",
              "sub_region_code": "202",
            },
          ]
    }
    locationData={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GovernanceIndicatorsStatistics should match the snapshot when the locationData array has length  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GovernanceIndicatorsStatistics
    dataBase={
        Object {
            "database_code": "WWGI",
            "name": "Worldwide Governance Indicators",
          }
    }
    dataSet={
        Object {
            "dataset_code": "VA",
            "name": "Voice and Accountability",
          }
    }
    location={
        Array [
            Object {
              "alpha_2": "BF",
              "alpha_3": "BFA",
              "country_code": "854",
              "intermediate_region": "Western Africa",
              "intermediate_region_code": "011",
              "iso_3166_2": "ISO 3166-2: BF",
              "name": "Burkina Faso",
              "region": "Africa",
              "region_code": "002",
              "sub_region": "Sub-Saharan Africa",
              "sub_region_code": "202",
            },
          ]
    }
    locationData={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
